### PR TITLE
add a result type

### PR DESCRIPTION
**Pull request by martindemello**
A very simple implementation of a result type that wraps and returns exceptions from Message and Shell rather than reraising them. Tested manually by introducing errors to the example/* files and making sure the vroom output didn't change.

I believe this fixes all instances of reraising other than in the main Run loop.

### CODE REVIEW

File: vroom/result.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# Inherit from namedtuple so we get an immutable value.
+class Result(namedtuple('Result', ['status', 'value'])):
+  """Holds the result or error of a function call.
+
+  status should be one of ResultType.result or ResultType.error
+  """
+
+  @classmethod
+  def Result(cls, value):
```
Comment (dbarnett) [inline vroom/result.py]: Can we just have a result.Success or something semantic like that? AFAICT you never actually read the True values and True is just a placeholder.

**Comment by dbarnett** [comment]
LGTM apart from one comment on the `Result(True)` idiom.

**Comment by martindemello** [comment]
Added. Starts fixing #74

**Comment by dbarnett** [comment]
LGTM.

**Pull request merged**